auth/internal/trustboundary/trust_boundary.go

```diff
@@ -0,0 +1,324 @@
+	GetUniverseDomain(ctx context.Context) (string, error)
+}
+
+// AllowedLocationsResponse is the structure of the response from the Trust Boundary API.
+type AllowedLocationsResponse struct {
+	// Locations is the list of allowed locations.
+	Locations []string `json:"locations"`
+	// EncodedLocations is the encoded representation of the allowed locations.
+	EncodedLocations string `json:"encodedLocations"`
+}
+
+// Data represents the trust boundary data.
+type Data struct {
+	// Locations is the list of locations that the token is allowed to be used in.
+	locations []string
+	// EncodedLocations represents the locations in an encoded format.
+	encodedLocations string
+}
+
+// Locations returns a read-only copy of the allowed locations.
+func (t *Data) Locations() []string {
+	if t == nil {
+		return nil
+	}
+	locs := make([]string, len(t.locations))
+	copy(locs, t.locations)
+	return locs
+}
+
+// EncodedLocations returns the encoded representation of the allowed locations.
+func (t *Data) EncodedLocations() string {
+	if t == nil {
+		return ""
+	}
+	return t.encodedLocations
+}
+
+// IsNoOpOrEmpty reports whether the trust boundary is a no-op or empty.
+// A no-op trust boundary is one where no restrictions are enforced.
+// An empty trust boundary is one where no locations are specified.
+func (t *Data) IsNoOpOrEmpty() bool {
+	if t == nil {
+		return true
+	}
+	return t.encodedLocations == NoOpEncodedLocations || t.encodedLocations == ""
+}
+
+// NewNoOpTrustBoundaryData returns a new TrustBoundaryData with no restrictions.
+func NewNoOpTrustBoundaryData() *Data {
+	return &Data{
+		encodedLocations: NoOpEncodedLocations,
+	}
+}
+
+// NewTrustBoundaryData returns a new TrustBoundaryData with the specified locations and encoded locations.
+func NewTrustBoundaryData(locations []string, encodedLocations string) *Data {
+	locationsCopy := make([]string, len(locations))
+	copy(locationsCopy, locations)
+	return &Data{
+		locations:        locationsCopy,
+		encodedLocations: encodedLocations,
+	}
+}
+
+// fetchTrustBoundaryData fetches the trust boundary data from the API.
+func fetchTrustBoundaryData(ctx context.Context, client *http.Client, url string, accessToken string) (*Data, error) {
+	if client == nil {
+		return nil, errors.New("trustboundary: HTTP client is required")
+	}
+
+	if url == "" {
+		return nil, errors.New("trustboundary: URL cannot be empty")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("trustboundary: failed to create trust boundary request: %w", err)
+	}
+
+	if accessToken == "" {
+		return nil, errors.New("trustboundary: access token required for lookup API authentication")
+	}
+	req.Header.Set("Authorization", "Bearer "+accessToken)
+
+	response, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("trustboundary: failed to fetch trust boundary: %w", err)
+	}
+	defer response.Body.Close()
+
+	body, err := io.ReadAll(response.Body)
```

Comment:
Can we plumb through the logger so that the details of this request can be logged if a use wants to see that information. You can see examples of how this is done in auth.go.